xlators/storage/posix/src/posix-handle.c

```diff
@@ -844,33 +844,6 @@ posix_handle_soft(xlator_t *this, const char *real_path, loc_t *loc,
                    "symlink %s -> %s failed", oldpath, newstr);
             return -1;
         }
-
-        ret = sys_fstatat(dfd, newstr, &newbuf, AT_SYMLINK_NOFOLLOW);
-
-        if (ret) {
-            gf_msg(this->name, GF_LOG_WARNING, errno, P_MSG_HANDLE_CREATE,
-                   "stat on %s failed ", newstr);
-            return -1;
-        }
-    }
-
-    ret = sys_stat(real_path, &newbuf);
-    if (ret) {
-        gf_msg(this->name, GF_LOG_WARNING, errno, P_MSG_HANDLE_CREATE,
-               "stat on %s failed ", real_path);
-        return -1;
-    }
-
-    if (!oldbuf)
-        return ret;
-
-    if (newbuf.st_ino != oldbuf->st_ino || newbuf.st_dev != oldbuf->st_dev) {
```

Comment:
We can't delete this check, it is required to validate real_path/gfid_path both are having same inode.

Reply:
Let me go through the code once more. It didn't look like that is what is happening for directories.

Reply:
Yes you are correct for directory it is checking inode number with same (real_path) so stat call is not required.